src/components/views/settings/TchapJoinRuleSettings.tsx

```diff
@@ -0,0 +1,330 @@
+        description: _t("Only invited people can join."),
+        checked: joinRule === JoinRule.Invite || (joinRule === JoinRule.Restricted && !restrictedAllowRoomIds?.length),
+    }, {
+        value: JoinRule.Public,
+        label: _t("Public"),
+        description: <>
+            { _t("Anyone can find and join.") }
+            { aliasWarning }
+        </>,
+    }];
+
+    if (roomSupportsRestricted || preferredRestrictionVersion || joinRule === JoinRule.Restricted) {
+        let upgradeRequiredPill;
+        if (preferredRestrictionVersion) {
+            upgradeRequiredPill = <span className="mx_JoinRuleSettings_upgradeRequired">
+                { _t("Upgrade required") }
+            </span>;
+        }
+
+        let description;
+        if (joinRule === JoinRule.Restricted && restrictedAllowRoomIds?.length) {
+            // only show the first 4 spaces we know about, so that the UI doesn't grow out of proportion there are lots.
+            const shownSpaces = restrictedAllowRoomIds
+                .map(roomId => cli.getRoom(roomId))
+                .filter(room => room?.isSpaceRoom())
+                .slice(0, 4);
+
+            let moreText;
+            if (shownSpaces.length < restrictedAllowRoomIds.length) {
+                if (shownSpaces.length > 0) {
+                    moreText = _t("& %(count)s more", {
+                        count: restrictedAllowRoomIds.length - shownSpaces.length,
+                    });
+                } else {
+                    moreText = _t("Currently, %(count)s spaces have access", {
+                        count: restrictedAllowRoomIds.length,
+                    });
+                }
+            }
+
+            const onRestrictedRoomIdsChange = (newAllowRoomIds: string[]) => {
+                if (!arrayHasDiff(restrictedAllowRoomIds || [], newAllowRoomIds)) return;
+
+                if (!newAllowRoomIds.length) {
+                    setContent({
+                        join_rule: JoinRule.Invite,
+                    });
+                    return;
+                }
+
+                setContent({
+                    join_rule: JoinRule.Restricted,
+                    allow: newAllowRoomIds.map(roomId => ({
+                        "type": RestrictedAllowType.RoomMembership,
+                        "room_id": roomId,
+                    })),
+                });
+            };
+
+            const onEditRestrictedClick = async () => {
+                const restrictedAllowRoomIds = await editRestrictedRoomIds();
+                if (!Array.isArray(restrictedAllowRoomIds)) return;
+                if (restrictedAllowRoomIds.length > 0) {
+                    onRestrictedRoomIdsChange(restrictedAllowRoomIds);
+                } else {
+                    onChange(JoinRule.Invite);
+                }
+            };
+
+            /**
+             * TCHAP : disable space-related options if create space feature is not enabled
```

Comment:
Hmmm, we are losing future evolutions of element's JoinRuleSettings.tsx... 🤔 Maybe @jdauphant 's idea of using CSS to hide things would be easier to maintain... 
But this works and moves forward. We can try the CSS way in another PR.

The e2e test is the same in both cases (yay e2e test !)

Reply:
Other idea : instead of customisations.json, use patch-package to add the `if` into element's JoinRuleSettings.tsx. 

It would not be too hard to maintain : at each upgrade of react-sdk, `yarn install` triggers a patch-package warning, like "the patch was made for vX.X.X and you are now at v.Y.Y.Y, no garantees". And if the change is too great, the patching breaks and yarn install fails.

And the e2e test is STILL the same !